src/engine/data_gen.c

```diff
@@ -124,6 +181,27 @@ data_gen_sample_create(Memory* memory, DataGen* data, u32 idx) {
         }
         sample_pulses->next_pulse_time = data_pulses->first_pulse_time;
         sample_pulses->next_between_pulses_time = data_pulses->first_pulse_time + data_pulses->pulse_duration;
+    } else if (data->type == DATA_GEN_ROC) {
+        sample->type = DATA_SAMPLE_ROC;
+
+        DataGenRoc* data_roc = &(data->roc);
+        DataSampleRoc* sample_roc = &(sample->roc_spikes);
+
+        if (data_roc->current_sample == NULL) {
+            log_error("NO MORE SAMPLES. Shouldn't be called this much");
+        }
+        String* sample_path = string_path_join(memory, data_roc->encodings_path, data_roc->current_sample->name);
+        data_roc->current_sample = data_roc->current_sample->next;
+
+        InputSpikeTimes* spike_times = input_spike_times_read(memory, string_get_c_str(sample_path));
+        check_memory(spike_times);
+        // TODO: daca ai duration mai mare ca time sa pui duration in loc la time
```

Comment:
This is for when the sample is not as long as the desired sample duration right?
I think I fix this below, but check
Also it may be an idea to use this in the matrix directly, we will se